akka-javasdk/src/main/scala/akka/javasdk/impl/agent/ToolDescriptors.scala

```diff
@@ -17,13 +17,10 @@ import akka.runtime.sdk.spi.SpiAgent
 @InternalApi
 object ToolDescriptors {
 
-  def apply(any: Any): Seq[SpiAgent.ToolDescriptor] =
-    apply(any.getClass)
-
   def agentToolDescriptor(cls: Class[_]): Seq[SpiAgent.ToolDescriptor] =
     toToolDescriptors(cls, allowNonPublic = true)
 
-  def apply(cls: Class[_]): Seq[SpiAgent.ToolDescriptor] =
+  def forClass(cls: Class[_]): Seq[SpiAgent.ToolDescriptor] =
     toToolDescriptors(cls, allowNonPublic = false)
 
   private def toToolDescriptors(cls: Class[_], allowNonPublic: Boolean): Seq[SpiAgent.ToolDescriptor] = {
```

Comment:
I wonder if we should inform a user that the provided tool class does not contain any tools. A common issue could be a method that is not public or missing an annotation. Maybe we could log that fact?